portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,20 +13,69 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
+import com.google.gson.Gson;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
 
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.*; 
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returnd recommendations data*/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+ 
+  /**
+  * do request which return as response all the recommendations ap to given limit
+  */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
I know that this doGet name was here before but it is not such a good name. Try to choose a name so it will be easier to understand what the function is doing